File: backend/src/main/java/com/gudominguete/desafiofundecc/models/Usuario.java
package com.gudominguete.desafiofundecc.models;

import jakarta.persistence.*;

import java.util.Date;
import java.util.List;

@Entity
@Table(name="usuario")
public class Usuario extends Pessoa{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "cargo_id")
    private Cargo cargo;

    @ManyToMany
    @JoinTable(
            name = "rel_usuario_perfil",
            joinColumns = @JoinColumn(name = "usuario_id"),
            inverseJoinColumns = @JoinColumn(name = "perfil_id"))
    private List<Perfil> perfis;

    public Cargo getCargo() {
        return cargo;
    }

    public void setCargo(Cargo cargo) {
        this.cargo = cargo;
    }

    public List<Perfil> getPerfis() {
        return perfis;
    }

    public void setPerfis(List<Perfil> perfis) {
        this.perfis = perfis;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
}

File: frontend/src/app/usuario/usuario.ts
import { Cargo } from "../cargos/cargo";
import { Perfil } from "../perfil/perfil";

export class Usuario {
    id: number;
    nome: string;
    cpf: string;
    sexo: string;
    dataNascimento: Date | null;
    cargo: Cargo | null;
    perfis: Perfil[];

    constructor(){
        this.id = 0;
        this.nome = "";
        this.cpf = "";
        this.sexo = "";
        this.dataNascimento = null;
        this.cargo = null;
        this.perfis = [];
    }
}
